docs/user_guide/shifts.md

```diff
@@ -0,0 +1,203 @@
+:dedent:
+:language: python
+:start-at: '# add column aliases for shift mu'
+:end-before: '# external files'
+```
+
+Note that this function makes some assumptions on the naming of the varied quantities and the shifts.
+
+Finally, we can now configure the modules within the workflow to use this shift.
+As mentioned above, this class of systematic uncertainties does not change any selection efficiencies.
+Therefore, it is sufficient to consider the varied quantities in the {py:class}`~columnflow.tasks.histograms.CreateHistograms` task.
+
+The calculation of the final event weight is handles by the {py:class}`~columnflow.weight.WeightProducer` instance you specify at command-line level.
+Consequently, this module needs to be aware of the shift.
+This can be done with the internal {py:attr}`~columnflow.columnar_util.TaskArrayFunction.shifts` set, see e.g. the analysis example for a weight producer:
+
+```{literalinclude} ../../analysis_templates/cms_minimal/__cf_module_name__/weight/example.py
+:dedent:
+:language: python
+:pyobject: example_init
+```
+
+Here, we use the {py:func}`~columnflow.config_util.get_shifts_from_sources` function to extract all shifts for source `mu` from the config inst.
+
+Adding a source of uncertainty to this set denotes that this module is sensitive to this shift.
+Whenever a shift that we defined above is requested, the WeightProducer will internally switch to the column defined in the `column_alias` auxiliary information that we defined earlier.
+This way, the workflow is now fully aware of the shift.
+You can test your implementation e.g. by running the {py:class}`~columnflow.tasks.plotting.PlotShiftedVariables1D` task.
+For more information, see the [plotting overview](../task_overview/plotting_tasks.md).
+
+To include this shift in a statistical inference model, we need to add it as a parameter (see CMS analysis example):
+
+```{literalinclude} ../../analysis_templates/cms_minimal/__cf_module_name__/inference/example.py
+:dedent:
+:language: python
+:start-at: '# muon weight uncertainty'
+:end-before: '# jet energy correction uncertainty'
+```
+
+In particular, note that this type of uncertainty needs to define the keyword argument `config_shift_source` in the {py:func}`~columnflow.inference.InferenceModel.add_parameter` function to properly resolve the necessary shifts to run through the workflow.
+
+## Uncertainties that modify the selection efficiency
+
+A more complex type of shape uncertainty arises from effects that can modify vital quantities for analysis decisions.
+One example for this class of uncertainties are calibrations of objects such as jets, which modify the four-momenta content of these objects.
+Since these calibrations are generally subject to uncertainties, the corresponding variations can propagate throughout the analysis workflow and modify for example selection efficiencies in the analysis phase space.
```

Comment:
```suggestion
Since these calibrations are generally subject to uncertainties, the corresponding variations can propagate throughout the analysis workflow and modify event and object selection acceptances.
```